src/output.ts

```diff
@@ -172,29 +188,41 @@ export default class Output {
 
 				for (let line of lines) {
 					const currentLine = output[y + offsetY];
-
 					// Line can be missing if `text` is taller than height of pre-initialized `this.output`
-					if (!currentLine) {
-						continue;
-					}
-
-					const width = stringWidth(line);
+					if (!currentLine) continue;
 
 					for (const transformer of transformers) {
 						line = transformer(line);
 					}
 
-					output[y + offsetY] =
-						sliceAnsi(currentLine, 0, x) +
-						line +
-						sliceAnsi(currentLine, x + width);
+					const chars = styledCharsFromTokens(tokenize(line));
+					let offsetX = x;
+					for (const char of chars) {
+						currentLine[offsetX] = char;
+
+						// Some characters take up more than one column. In that case, the following
+						// pixels need to be cleared to avoid printing extra characters
+						const charWidth = char.fullWidth ? 2 : char.value.length;
+						for (let dx = 1; dx < charWidth; dx++) {
```

Comment:
Why does this skip the first character and starts from the second one?

Reply:
Because the first char is the char that's actually written, so width 1 is already accounted for.

Reply:
Not sure I understand, could you elaborate the "that's actually written" part?

Reply:
Line 201 writes the current character to the output array. Some characters are wider than a single column though, so we have to set the following columns (up to the character width) to "" so we don't end up with extra characters.
So if the char is a 2-column emoji, we write the emoji to column x and "" to column x+1. The character after the emoji will then correctly be at column x+2.

Reply:
Do we need a loop here then? Could this be simplified to this?

```js
if (char.fullWidth) {
  currentLine[offsetX + 1] = /* empty char */
}

offsetX += character.fullWidth ? 2 : 1
```

Reply:
Aren't there some emojis which have length > 4?

Reply:
I've pushed a commit to eliminate the loop.